libbpf-rs/src/iter.rs

```diff
@@ -0,0 +1,57 @@
+use nix::{libc, unistd};
+use std::io;
+
+use crate::*;
+
+/// Represents a bpf iterator for reading kernel data structures. This requires
+/// Linux 5.8.
+pub struct Iter {
+    link: Link,
+}
+
+impl Iter {
+    pub fn new(link: Link) -> Self {
+        Self { link }
+    }
+
+    /// Open the iterator for reading.
+    pub fn open(&self) -> Result<OpenIter> {
+        let link_fd = self.link.get_fd();
+        let fd = unsafe { libbpf_sys::bpf_iter_create(link_fd) };
+        if fd < 0 {
+            return Err(Error::System(-fd));
+        }
+        Ok(OpenIter { fd })
+    }
```

Comment:
Any reason we can't do the contents of `open()` inside `Iter::new()`? When `Iter` is dropped we can close the fd there.

Reply:
I'm trying to imagine why someone would create an `Iter` but not call `iter.open()` on it but I can't think of any reason.

Reply:
Right. Yeah I guess there isn't really a use case for calling `open()` multiple times either. Let's remove it.